Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -179,24 +186,26 @@ namespace ROS2
     }
 
     void ROS2SpawnerComponent::GetSpawnPointsNames(
-        const ROS2::GetSpawnPointsNamesRequest request, ROS2::GetSpawnPointsNamesResponse response)
+        const ROS2::GetSpawnPointsNamesRequest& request, ROS2::GetSpawnPointsNamesResponse response)
     {
-        for (auto spawnPoint : GetSpawnPoints())
+        for (const auto& [name, info] : GetSpawnPoints())
```

Comment:
GetSpawnPoints is currently checking only children for SpawnPointComponent, what do you think about changing it to all descendants? Use case I have in mind: AMR docking station as a prefab - it could provide a spawning point right off the shelf